File: java/event-processor-host/src/main/java/com/microsoft/azure/eventprocessorhost/Lease.java
/*
 * Copyright (c) Microsoft. All rights reserved.
 * Licensed under the MIT license. See LICENSE file in the project root for full license information.
 */

package com.microsoft.azure.eventprocessorhost;

public class Lease
{
    private String partitionId;

    private long epoch;
    private String owner;
    private String token;

    public Lease(String partitionId)
    {
        this.partitionId = partitionId;

        this.epoch = 0;
        this.owner = "";
        this.token = "";
    }

    public Lease(Lease source)
    {
        this.partitionId = source.partitionId;

        this.epoch = source.epoch;
        this.owner = source.owner;
        this.token = source.token;
    }

    public long getEpoch()
    {
        return this.epoch;
    }

    public void setEpoch(long epoch)
    {
        this.epoch = epoch;
    }
    
    public long incrementEpoch()
    {
    	this.epoch++;
    	return this.epoch;
    }
    
    public String getOwner()
    {
        return this.owner;
    }

    public void setOwner(String owner)
    {
        this.owner = owner;
    }

    public String getPartitionId()
    {
        return this.partitionId;
    }

    public String getToken()
    {
        return this.token;
    }

    public void setToken(String token)
    {
        this.token = token;
    }

    public boolean isExpired() throws Exception
    {
    	// this function is meaningless in the base class
    	return false;
    }
}

File: csharp/src/Microsoft.Azure.EventHubs.Processor/Lease.cs
// Copyright (c) Microsoft. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

namespace Microsoft.Azure.EventHubs.Processor
{
    using System;
    using Newtonsoft.Json;

    public class Lease
    {
        protected Lease()
        {
        }

        protected Lease(string partitionId)
        {
            this.PartitionId = partitionId;
            this.Owner = string.Empty;
            this.Token = string.Empty;
        }

        protected Lease(Lease source)
        {
            this.PartitionId = source.PartitionId;
            this.Epoch = source.Epoch;
            this.Owner = source.Owner;
            this.Token = source.Token;
        }

        public string PartitionId { get; set; }

        public string Owner { get; set; }

        public string Token { get; set; }

        public long Epoch { get; set; }

        public virtual bool IsExpired() 
        {
    	    // this function is meaningless in the base class
    	    return false;
        }

        public virtual string GetStateDebug()
        {
            return "N/A";
        }

        internal long IncrementEpoch()
        {
            this.Epoch++;
            return this.Epoch;
        }
    }
}